opmon-protobuf-dbwriter/dbwriter.py

```diff
@@ -0,0 +1,168 @@
+
+#influx options
+@click.option('--influxdb-address', type=click.STRING, default='np04-srv-017', help='address of the influx db')
+@click.option('--influxdb-port',    type=click.INT,    default=31002,          help='port of the influxdb')
+@click.option('--influxdb-name',    type=click.STRING, default='test_influx',  help='Table name destination inside influxdb')
+@click.option('--influxdb-create',  type=click.BOOL,   default=True,           help='Creates the influxdb if it does not exists')
+@click.option('--influxdb-timeout', type=click.INT,    default=500,            help='Size in ms of the batches sent to influx')
+
+@click.option('--debug',       type=click.BOOL, default=True, help='Set debug print levels')
+
+def cli(subscriber_bootstrap, subscriber_group, subscriber_timeout, subscriber_topic,
+        influxdb_address, influxdb_port, influxdb_name, influxdb_create, influxdb_timeout,
+        debug):
+
+    logging.basicConfig(
+        format='%(asctime)s %(levelname)-8s %(message)s',
+        level=logging.DEBUG if debug else logging.INFO,
+        datefmt='%Y-%m-%d %H:%M:%S')
+
+    influx = InfluxDBClient(host=influxdb_address, port=influxdb_port)
+    db_list = influx.get_list_database()
+    logging.info("Available DBs:",db_list)
+    if {"name":influxdb_name}  not in db_list:
+        logging.warning(influxdb_name, "DB not available")
+        if influxdb_create:
+            influx.create_database(influxdb_name);
+            logging.info("New list of DBs:", influx.get_list_database())
+
+    influx.switch_database(influxdb_name)
+
+    sub = opmon_sub.OpMonSubscriber( bootstrap=subscriber_bootstrap,
+                                     topics=subscriber_topic,
+                                     group_id = subscriber_group,
+                                     timeout_ms = subscriber_timeout)
+
+    # this is a list of Entries
+    q = queue.Queue()
+
+    callback_function = partial(process_entry, 
+                                q = q )
+    
+    sub.add_callback(name="to_influx", 
+                     function=callback_function)
+
+    thread = threading.Thread(target=consume, daemon=True,
+                              args=(q,influxdb_timeout,influx) )
+    thread.start()
+    
+    sub.start()
+
+def consume( q : queue.Queue, timeout_ms,
+             influx : InfluxDBClient = None ) :
+    logging.info("Starting consumer thread")
+    batch=[]
+    batch_ms = 0
+    while True :
+        try :
+            entry = q.get(timeout=1) ## timeout here is in seconds
+
+            if ( entry.ms - batch_ms < timeout_ms ) :
+                batch.append(entry.json)
+                batch_ms = min(batch_ms, entry.ms)
+                
+            if ( entry.ms - batch_ms >= timeout_ms ) :
+                send_batch(batch)
+                batch=[entry.json]
+                batch_ms = entry.ms
+            
+        except queue.Empty :
+            logging.debug("Queue is empty")
```

Comment:
I'd be tempted to make this log at `info` level.  But you know the code/workflow better than I.  If you think `debug` is the right level that is good enough for me.

Reply:
I think this is ok otherwise this will create a huge log and for nothing really.